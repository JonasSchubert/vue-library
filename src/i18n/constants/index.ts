export const apiUrl: string = 'http://192.168.178.39:5099/i18n-service/api/v1/';
