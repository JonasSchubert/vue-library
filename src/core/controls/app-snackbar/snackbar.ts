import { EventBus } from '@/core/bus';
import { SnackbarConfig } from './snackbar-config';
import { SnackbarEvents } from './snackbar-events.enum';
import { defaultSnackbarConfig } from './snackbar.constants';

export const snackbar = {
  error: (config: SnackbarConfig) => {
    EventBus.$emit(SnackbarEvents.show, {
      ...defaultSnackbarConfig,
      color: 'error',
      icon: 'mdi-alert-circle',
      iconColor: 'white',
      ...config
    });
  },
  hide: () => EventBus.$emit(SnackbarEvents.hide),
  info: (config: SnackbarConfig) => {
    EventBus.$emit(SnackbarEvents.show, {
      ...defaultSnackbarConfig,
      color: 'info',
      icon: 'mdi-information',
      iconColor: 'white',
      ...config
    });
  },
  open: (config: SnackbarConfig) => {
    EventBus.$emit(SnackbarEvents.show, {
      ...defaultSnackbarConfig,
      ...config
    });
  },
  success: (config: SnackbarConfig) => {
    EventBus.$emit(SnackbarEvents.show, {
      ...defaultSnackbarConfig,
      color: 'success',
      icon: 'mdi-check',
      iconColor: 'white',
      ...config
    });
  },
  warning: (config: SnackbarConfig) => {
    EventBus.$emit(SnackbarEvents.show, {
      ...defaultSnackbarConfig,
      color: 'warning',
      icon: 'mdi-alert',
      iconColor: 'white',
      ...config
    });
  }
};
