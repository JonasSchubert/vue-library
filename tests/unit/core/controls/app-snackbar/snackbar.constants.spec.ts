import { defaultSnackbarConfig } from '@/core/controls/app-snackbar/snackbar.constants';

describe('defaultSnackbarConfig', () => {
  test('should not change unexpected', () => {
    // Assert
    expect(defaultSnackbarConfig).toMatchSnapshot();
  });
});
