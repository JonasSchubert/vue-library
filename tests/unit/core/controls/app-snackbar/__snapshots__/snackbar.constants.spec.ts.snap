// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`defaultSnackbarConfig should not change unexpected 1`] = `
Object {
  "btn": undefined,
  "color": "primary",
  "icon": "",
  "iconColor": "accent",
  "message": "",
  "messageParams": Array [],
  "multiLine": false,
  "timeout": 3000,
  "translate": true,
  "vertical": false,
}
`;
